pkg/component/watch.go

```diff
@@ -120,8 +145,24 @@ func WatchAndPush(client *occlient.Client, componentName string, applicationName
 						break
 					}
 				}
-				if !alreadyInChangedFiles {
-					changedFiles = append(changedFiles, event.Name)
+
+				// Filter out anything in ignores list from the list of changed files
+				// This is important inspite of not watching the
+				// ignores paths because, when a directory that is ignored, is deleted,
+				// because its parent is watched, the fsnotify automatically raises an event
+				// for it.
+				matched, err := isRegExpMatch(event.Name, ignores)
+				if err != nil {
+					watchError = errors.Wrap(err, "unable to watch changes")
+				}
+				if !alreadyInChangedFiles && !matched {
+					// In windows, every new file created under a sub-directory of the watched directory, raises 2 events:
+					// 1. Write event for the directory under which the file was created
+					// 2. Create event for the file that was created
+					// Ignore 1 to avoid duplicate events.
+					if !(stat.IsDir() && event.Op&fsnotify.Write == fsnotify.Write) {
```

Comment:
how does `event.Op&fsnotify.Write == fsnotify.Write` work?

Reply:
Not sure :)
Just followed the one from https://github.com/redhat-developer/odo/blob/master/pkg/component/watch.go#L129
Did not venture into that detail(in part because of the urgency of the PR)